Feat:Create llama how-tos

tomek-labuk opened the pull request:
## Description

Adds missing and simple ollama how-tos for AI Proxy and AI Proxy Advanced plugins.

## Preview Links


## Checklist 

- [x] Every page is page one.
- [x] Tested how-to docs. If not, note why here. 
- [x] All pages contain metadata.
- [ ] Updated [sources.yaml](https://github.com/Kong/developer.konghq.com/blob/main/tools/track-docs-changes/config/sources.yml). For more info, review [track docs changes](https://github.com/Kong/developer.konghq.com/tree/main/tools/track-docs-changes)
- [x] Any new docs link to existing docs.
- [x] All autogenerated instructions render correctly (API, decK, Konnect, Kong Manager). 
- [x] Style guide (capitalized gateway entities, placeholder URLs) implemented correctly.
- [x] Every page has a `description` entry in frontmatter
- [x] Add new pages to the product documentation index (if applicable)

cloudjumpercat commented on app/_includes/prereqs/ollama.md:
```diff
@@ -0,0 +1,32 @@
+
+1. Visit the [Ollama download page](https://ollama.com/download) and download the installer for your operating system. Follow the installation instructions for your platform.
+
+2. After installation, open your terminal and run:
```
````suggestion
1. Start Ollama:
   ```sh
   ollama start
   ```

2. After installation, open a new terminal window and run:
````
I had to start it before that model command would work. I installed via homebrew.

cloudjumpercat commented on app/_how-tos/set-up-ai-proxy-advanced-with-ollama.md:
```diff
@@ -0,0 +1,91 @@
+                    upstream_url: ${ollama_upstream_url}/api/chat
+variables:
+  ollama_upstream_url:
+    value: $ANTHROPIC_API_KEY
```
```suggestion
    value: $OLLAMA_UPSTREAM_URL
```
Also, we exported this variable in the prereq.

cloudjumpercat commented on app/_includes/prereqs/ollama.md:
```diff
@@ -0,0 +1,32 @@
+
+   Replace `<model-name>` with the model you want to run, such as `llama2` or `llama3`.
+
+3. To set up the AI Proxy plugin, you'll need the upstream URL of your local Llama instance. By default, Ollama runs at `localhost:11434`. You can verify this by running:
```
````suggestion
3. To set up the AI Proxy plugin, you'll need the upstream URL of your local Llama instance. 

   ```sh
   export DECK_OLLAMA_UPSTREAM_URL='http://localhost:11434'
   ```


   By default, Ollama runs at `localhost:11434`. You can verify this by running:
````

cloudjumpercat commented on app/_how-tos/set-up-ai-proxy-advanced-with-ollama.md:
```diff
@@ -0,0 +1,91 @@
+    value: $ANTHROPIC_API_KEY
+{% endentity_examples %}
+
+{:.info}
+> If you're running {{site.base_gateway}} locally in a Docker container, set `upstream_url` to `http://host.docker.internal:11434/api/chat`
```
```suggestion
```
Moving this note to the prereqs

cloudjumpercat commented on app/_includes/prereqs/ollama.md:
````diff
@@ -0,0 +1,32 @@
+   sudo lsof -iTCP -sTCP:LISTEN -n -P
+   ```
+
+   - Then look for the `ollama` process in the output and note the port number it’s listening on.
````
```suggestion
   - Then look for the `ollama` process in the output and note the port number it’s listening on.
   
   {:.info}
   > If you're running {{site.base_gateway}} locally in a Docker container, export your upstream URL as `http://host.docker.internal:11434`
```

cloudjumpercat commented on app/_how-tos/set-up-ai-proxy-with-ollama.md:
```diff
@@ -0,0 +1,90 @@
+
+tags:
+  - ai
+  - openai
```
```suggestion
  - ollama
```

cloudjumpercat commented on app/_includes/prereqs/ollama.md:
````diff
@@ -0,0 +1,32 @@
+   ollama run <model-name>
+   ```
+
+   Replace `<model-name>` with the model you want to run, such as `llama2` or `llama3`.
````
```suggestion
   You can replace `llama2` with the model you want to run, such as `llama3`.
```

cloudjumpercat reviewed (changes requested):
A few suggestions:
- Add links to these how tos from their Ollama plugin example counterparts on the AI Proxy (Advanced) plugin example pages
- The upstream URL export was missing from the prereqs/steps. I added a suggestion for it as well as a few other suggested revisions to the Ollama prereq.

tomek-labuk commented on app/_how-tos/set-up-ai-proxy-with-ollama.md:
```diff
@@ -0,0 +1,90 @@
+---
+title: Set up AI Proxy with Ollama
```
Adding it now.

tomek-labuk commented:
> A few suggestions:
> 
> * Add links to these how tos from their Ollama plugin example counterparts on the AI Proxy (Advanced) plugin example pages
> * The upstream URL export was missing from the prereqs/steps. I added a suggestion for it as well as a few other suggested revisions to the Ollama prereq.

All fixed + I've added references to these ho-tos on the AI Providers page.

cloudjumpercat commented on app/_how-tos/set-up-ai-proxy-advanced-with-ollama.md:
```diff
@@ -0,0 +1,90 @@
+
+## Validate
+
+{% include how-tos/steps/ai-proxy-validate.md %}
```
For some reason, this validate step is now giving me no output. I can't seem to figure out what's wrong.

cloudjumpercat commented on app/_how-tos/set-up-ai-proxy-with-ollama.md:
```diff
@@ -0,0 +1,88 @@
+
+## Validate
+
+{% include how-tos/steps/ai-proxy-validate.md %}
```
Same with this one, I don't get any output from this now.

cloudjumpercat reviewed (commented):
Just had some problems with the validate step, I've left comments for those.

cloudjumpercat commented on app/_how-tos/set-up-ai-proxy-advanced-with-ollama.md:
```diff
@@ -0,0 +1,90 @@
+                name: llama2
+                options:
+                    llama2_format: ollama
+                    upstream_url: ${ollama_upstream_url}/api/chat
```
I think I realized what went wrong from trying to do this in one of my PRs. decK won't accept content after the variable here. So `/api/chat` will have to be part of the $OLLAMA_UPSTREAM_URL.
```suggestion
                    upstream_url: ${ollama_upstream_url}
```

tomek-labuk replied:
Ah, there you go. Let me make this explicit in the prereq note!

cloudjumpercat commented on app/_includes/prereqs/ollama.md:
````diff
@@ -0,0 +1,47 @@
+3. To set up the AI Proxy plugin, you'll need the upstream URL of your local Llama instance. 
+
+   ```sh
+   export DECK_OLLAMA_UPSTREAM_URL='http://localhost:11434'
````
```suggestion
   export DECK_OLLAMA_UPSTREAM_URL='http://localhost:11434/api/chat'
```
Mentioned it down the page, but the `/api/chat` will have to be part of the variable for it to work in decK

cloudjumpercat commented on app/_how-tos/set-up-ai-proxy-with-ollama.md:
```diff
@@ -0,0 +1,88 @@
+          name: llama2
+          options:
+            llama2_format: ollama
+            upstream_url: ${ollama_upstream_url}/api/chat
```
```suggestion
            upstream_url: ${ollama_upstream_url}
```
